source/adapters/hip/device.cpp

```diff
@@ -493,8 +546,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     // name instead, this is also what AMD OpenCL devices return.
     if (strlen(Name) == 0) {
       hipDeviceProp_t Props;
-      detail::ur::assertion(hipGetDeviceProperties(&Props, hDevice->get()) ==
-                            hipSuccess);
+      if (hipGetDeviceProperties(&Props, hDevice->get()) != hipSuccess) {
```

Comment:
Should we make some effort to try and translate native error returns to UR specific? There is a bunch of native calls (here and in CUDA: `cuMemGetInfo`, `cuDeviceTotalMem` etc.) that just hard code `UR_RESULT_ERROR_INVALID_OPERATION`/`UR_RESULT_ERROR_INVALID_SIZE` on failure. Previously they were wrapped up in assertion, so the value didn't really matter, but now, they will be served up all the way to the user.